### PR TITLE
Fix outlier handling in qap when len(seeds)==n

### PR DESCRIPTION
**Pull request by kellymarchisio**
- [ ] Does this PR add any new dependencies? No
- [ ] Does this PR modify any existing APIs? No
   - [ ] Is the change to the API backwards compatible? Yes
- [ ] Have you built the documentation (reference and/or tutorial) and verified the generated documentation is appropriate? N/A

#### What does this implement/fix? Briefly explain your changes.
QAP incorrectly assumes the partial_match is sorted when partial_match.shape[0] == A.shape[0]. Fix this.

### CODE REVIEW

**Comment by bdpedigo** [comment]
although when `partial_match.shape[0] == A.shape[0]`, then QAP shouldn't run at all... is that the behavior you are expecting @kellymarchisio ?

**Comment by asaadeldin11** [comment]
@kellymarchisio Thanks for the PR! I think I understand the issue; the score returned is correct, but the `perm_inds` returned isn't necessarily correct, depending on the order of the first column of `partial_match`?

File: graspologic/match/qap.py [hunk 1]
```diff
@@ -400,6 +400,8 @@ def _quadratic_assignment_faq(
 
     # check outlier cases
     if n == 0 or partial_match.shape[0] == n:
+        # Cannot assume partial_match is sorted.
+        partial_match = np.row_stack(sorted(partial_match, key=lambda x: x[0]))
```
Comment (asaadeldin11) [inline graspologic/match/qap.py:404]: does sorted require partial_match to be a list?

Comment (kellymarchisio) [reply graspologic/match/qap.py:404]: Nope - it can be any iterable.

Comment (asaadeldin11) [reply graspologic/match/qap.py:404]: great. can you add a test to the function `test_barycenter_SGM` in test_match.py that gives (a possibly shuffled) full `partial_match` and returns the correct `score_` and `perm_inds`? (Should be very similar to the test lines 138-142)

**Comment by kellymarchisio** [comment]
Sure, done :)

**Review by asaadeldin11 — CHANGES_REQUESTED** [review]

File: tests/test_match.py [hunk 2]
```diff
@@ -141,6 +141,24 @@ def test_barycenter_SGM(self):
...
+        score = chr12c.score_
+        assert 11156 == score
+
+        W1 = np.array(range(n))
+        W2 = [pi[z] for z in W1]
+        # Shuffle seed pairs.
+        pairs = [i for i in zip(W1, W2)]
+        random.shuffle(pairs)
+        W1 = list(list(zip(*pairs))[0])
+        W2 = list(list(zip(*pairs))[1])
```
[the remark this answers is not in the document]
Comment (kellymarchisio) [reply tests/test_match.py:156]: haha oh boy, yep -- I'd been doing a lot of work with tuples earlier in the day so I was primed for my complicated answer XD. thanks!

**Review by asaadeldin11 — APPROVED** [review]

**Comment by asaadeldin11** [comment]
@bdpedigo this looks good to me

**Pull request merged**